src/test/java/com/solr/clientwrapper/solrwrapper/SolrCollectionTest.java

```diff
@@ -0,0 +1,225 @@
+package com.solr.clientwrapper.solrwrapper;
+
+import com.solr.clientwrapper.IntegrationTest;
+import com.solr.clientwrapper.domain.dto.solr.SolrResponseDTO;
+import com.solr.clientwrapper.domain.dto.solr.collection.SolrCreateCollectionDTO;
+import com.solr.clientwrapper.domain.dto.solr.collection.SolrDeleteCollectionDTO;
+import com.solr.clientwrapper.domain.dto.solr.collection.SolrGetCollectionsResponseDTO;
+import com.solr.clientwrapper.domain.dto.solr.collection.SolrRenameCollectionDTO;
+import com.solr.clientwrapper.domain.service.SolrCollectionService;
+import com.solr.clientwrapper.solrcore.TestUtil;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
+import org.springframework.security.test.context.support.WithMockUser;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+import org.springframework.transaction.annotation.Transactional;
+
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@IntegrationTest
+@AutoConfigureMockMvc
+@WithMockUser
+public class SolrCollectionTest {
+
+    String solrCollectionEndpoint ="/searchservice/table";
+
+    String collectionName ="automatedTestCollection";
+
+
+    @Autowired
+    private MockMvc restAMockMvc;
+
+    @MockBean
+    private SolrCollectionService solrCollectionService;
```

Comment:
Here, Controller layer is tested by mocking service layer. I believe, we have to **_test service layer methods_** too perhaps. Please see if it is required.

Reply:
Yes, only the controller layer and the data flow from controller to service layer are tested. Mocked services since the service layer contain code that needs the solr instance to be running. Running a Solr instance during CI CD pipeline is not possible hence this is done. The previous code used to test the service layer but because of the mentioned reason, I had to change the Test cases and mock service layer